Test/email

Odonata971 opened the pull request:
close #17

Commit:
test: add unit tests for email service

Implements unit tests to verify email sending functionality.
Mocks email sender to simulate email dispatch.
Covers scenarios for successful sending, failure cases, and getInstance method.

Commit:
test: add unit tests for email service

Implements unit tests to verify email sending functionality.
Mocks email sender to simulate email dispatch.
Covers scenarios for successful sending, failure cases, and getInstance method.

Francois389 commented on src/test/java/sae/semestre/six/email/EmailServiceTest.java:
```diff
@@ -0,0 +1,103 @@
+
+    @BeforeEach
+    void setUp() {
+        // Rediriger la sortie standard pour capturer les messages
```
Use english.

Francois389 reviewed (changes requested):
Change just the comment from French to English, then 👍.
Also good test.

Commit:
test: update comments for clarity in test setup

Improves readability by translating comments from French to
English in the test setup method.

Francois389 reviewed (changes requested):
You have still one change to make (https://github.com/SAE-S6-Evolution-applicative/SAE-S6-legacy-software/pull/16/files#r2054018395)

Odonata971 committed:
chore: Change the comments in EmailTestService.java 

to be cohesive with the rest of the project that all new parts will be written in English

Francois389 reviewed (dismissed):
👍

Francois389 commented:
Please rebase
